ext/mixed/js/core.js

```diff
@@ -288,6 +293,45 @@ const yomichan = (() => {
             this.trigger('orphaned', {error});
         }
 
+        getTemporaryListenerResult(eventHandler, userCallback, timeout=null) {
+            let resolve;
+            let reject;
+            const listenerPromise = new Promise((_resolve, _reject) => {
+                resolve = _resolve;
+                reject = _reject;
+            });
+
+            if (eventHandler === chrome.runtime.onMessage) {
```

Comment:
I would omit this check and just rely on duck-typing. If we include this type, we may as well force eventHandler to be chrome.runtime.onMessage

Reply:
https://github.com/FooSoft/yomichan/pull/417/commits/34d9ac6ad9bae72c6ce87dc49d30bf5b42908111